pkg/controllers/cassandra/service/seedprovider/resource.go

```diff
@@ -0,0 +1,39 @@
+package seedprovider
+
+import (
+	v1alpha1 "github.com/jetstack/navigator/pkg/apis/navigator/v1alpha1"
+	serviceutil "github.com/jetstack/navigator/pkg/controllers/cassandra/service/util"
+	"github.com/jetstack/navigator/pkg/controllers/cassandra/util"
+	apiv1 "k8s.io/api/core/v1"
+)
+
+func ServiceForCluster(
+	cluster *v1alpha1.CassandraCluster,
+) *apiv1.Service {
+	return updateServiceForCluster(cluster, &apiv1.Service{})
+}
+
+func updateServiceForCluster(
+	cluster *v1alpha1.CassandraCluster,
+	service *apiv1.Service,
+) *apiv1.Service {
+	service = service.DeepCopy()
+	service = serviceutil.SetStandardServiceAttributes(cluster, service)
+	service.SetName(util.SeedProviderServiceName(cluster))
+	service.Spec.Type = apiv1.ServiceTypeClusterIP
+	service.Spec.ClusterIP = "None"
+	// Headless service should not require a port.
+	// But without it, DNS records are not registered.
+	// See https://github.com/kubernetes/kubernetes/issues/55158
+	service.Spec.Ports = []apiv1.ServicePort{
+		{
+			Port: 65535,
+		},
+	}
+	// This ensures that DNS names are published regardless of whether the
+	// Cassandra pod ReadinessProbes (listening on their CQL port).
+	// It won't handle CQL connections until it has successfully connected and
+	// negotiated with a seed host
+	service.Spec.PublishNotReadyAddresses = true
```

Comment:
See also https://github.com/kubernetes/examples/issues/89

Reply:
This field has only been promoted in 1.8+ - we should also set the `service.alpha.kubernetes.io/tolerate-unready-endpoints ` annotation, which works back to 1.5 or 1.6 or something :smile:

Awesome to know that it's been promoted to a field now though!

Reply:
Isn't that handled by API version conversion? Otherwise how can they add a field to an API without changing the version?

Reply:
Yep it is - but if the target API server is not aware of the field, it will silently drop it. So if you submit the manifest produced above to a 1.7 APIServer, because 1.7 has never contained this field, it will be silently dropped.

If on the other hand you submit this to a 1.8 API server, it will automatically 'convert' it by additionally setting the aforementioned annotation. But this requires that you submit it to a 1.8 API server. Thus, we should set the annotation too if we want to support older k8s clusters. We should set both the field and annotation so that when 1.10 comes out and drops support for the annotation, we already support 1.10 as we set the field :smile:

Reply:
Got it. Thanks @munnerz 
I hadn't thought it through and maybe I was assuming some sort of conversion on the client side too.

Reply:
Done.